core/google/cloud/streaming/transfer.py

```diff
@@ -370,18 +370,30 @@ def initialize_download(self, http_request, http):
 
         :type http: :class:`httplib2.Http` (or workalike)
         :param http: Http instance for this request.
+
+        :type range_bytes: tuple
+        :param range_bytes: Optional. Range of bytes to download.
         """
         self._ensure_uninitialized()
         url = http_request.url
         if self.auto_transfer:
-            end_byte = self._compute_end_byte(0)
-            self._set_range_header(http_request, 0, end_byte)
+            if range_bytes:
+                start_byte, end_byte = range_bytes
+                end_byte = end_byte - 1
+            else:
+                start_byte = 0
+                end_byte = self._compute_end_byte(0)
+            self._set_range_header(http_request, start_byte, end_byte)
             response = make_api_request(
                 self.bytes_http or http, http_request)
             if response.status_code not in self._ACCEPTABLE_STATUSES:
                 raise HttpError.from_response(response)
             self._initial_response = response
             self._set_total(response.info)
+            # when using `range_bytes` we need to reset total_size
```

Comment:
I'm not sure about doing this here, but let's update this a tiny bit.
```diff
+ # When using `range_bytes` we need to reset total_size,
+ # otherwise we get the total bytes of the file and the entire contents
+ # are downloaded.
- # when using `range_bytes` we need to reset total_size
- # otherwise we get the total of the file and the entire contents
- # are downloaded
```